### PR TITLE
[CDAP-14864] Support sort by creation time and add tests

### PR DESCRIPTION
**Pull request by anew**
Please see individual commits.

### CODE REVIEW

**Comment by anew** [comment]
Verification test succeeded: https://builds.cask.co/browse/CDAP-DBV2-1
Unit tests running here: https://builds.cask.co/browse/CDAP-DUT6788

File: cdap-elastic/src/main/java/co/cask/cdap/metadata/elastic/MetadataDocument.java [hunk 1]
```diff
@@ -85,7 +93,9 @@ public boolean equals(Object o) {
       return false;
     }
     MetadataDocument that = (MetadataDocument) o;
-    return Objects.equals(entity, that.entity) &&
+    return created == that.created &&
```
Comment (poornachandra) [inline cdap-elastic/src/main/java/co/cask/cdap/metadata/elastic/MetadataDocument.java]: `created` is of type Long, need to use `equals()` here for check.

File: cdap-data-fabric/src/test/java/co/cask/cdap/spi/metadata/MetadataStorageTest.java [hunk 2]
```diff
@@ -877,86 +884,98 @@ public void testSearchPagination() throws IOException {
...
+  }
+
+  private void testSortedSearch(MetadataStorage mds, Sorting sorting, List<MetadataRecord> sorted) throws IOException {
+    // search once with sort by creation time and validate that records are in order
+    SearchResponse response = mds.search(SearchRequest.of("*").setSorting(sorting).build());
+    Assert.assertEquals(sorted, response.getResults());
+
+    // search for a window of results
+    assertInOrder(mds, SearchRequest.of("*").setSorting(sorting).setOffset(0).setLimit(4).build(),
+                  sorted.get(0), sorted.get(1), sorted.get(2), sorted.get(3));
```
Comment (poornachandra) [inline cdap-data-fabric/src/test/java/co/cask/cdap/spi/metadata/MetadataStorageTest.java]: Can use sorted.sublist(0, 4) to reduce verbosity here and below

**Review by poornachandra — COMMENTED** [review]
Just a couple of minor comments

File: cdap-elastic/src/main/java/co/cask/cdap/metadata/elastic/MetadataDocument.java (hunk 1 above)
Comment (anew) [reply cdap-elastic/src/main/java/co/cask/cdap/metadata/elastic/MetadataDocument.java]: good catch! I regenerated this with IDEA, not sure what went wrong, probably I messed up.

File: cdap-data-fabric/src/test/java/co/cask/cdap/spi/metadata/MetadataStorageTest.java (hunk 2 above)
Comment (anew) [reply cdap-data-fabric/src/test/java/co/cask/cdap/spi/metadata/MetadataStorageTest.java]: if the method assertInOrder did not expect an array (varargs)

File: cdap-elastic/src/main/java/co/cask/cdap/metadata/elastic/MetadataDocument.java (hunk 1 above)
Comment (anew) [reply cdap-elastic/src/main/java/co/cask/cdap/metadata/elastic/MetadataDocument.java]: regenerated both hashCode and equals

File: cdap-data-fabric/src/test/java/co/cask/cdap/spi/metadata/MetadataStorageTest.java (hunk 2 above)
Comment (anew) [reply cdap-data-fabric/src/test/java/co/cask/cdap/spi/metadata/MetadataStorageTest.java]: ok, I introduced a variation that takes a list.

**Review by poornachandra — APPROVED** [review]

**Comment by anew** [comment]
build passed. squashed and rebased, merging now.

**Pull request merged**